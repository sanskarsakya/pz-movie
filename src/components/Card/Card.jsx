
import React from "react";
import { Link } from "react-router-dom";

function Card({ id, name, image, vote_average, popularity, year }) {
  return (
    <div className="w-1/5 p-3 transform hover:-translate-y-2 ease-in-out duration-200 transition-translate">
      <div
        className="overflow-hidden"
        style={{ height: 200, borderRadius: 20 }}
      // width:270
      >
        <img
          className="object-cover"
          src={"https://image.tmdb.org/t/p/w500" + image}
          alt=""
        />
      </div>
      <Link to={"/movie/" + id}>
        <p  data-testid="movie_name"  className="mt-5 text-xl font-semibold cursor-pointer">{name}</p>
      </Link>
      <div
        className="mt-1 text-sm tracking-wider flex items-center"
        style={{ color: "rgba(255, 255, 255, 0.8)" }}
      >
        <span className="mr-6">{year}</span>
        <p>Adventure, Comedy</p>
      </div>
      <div className="mt-3 text-sm tracking-wider flex items-center">
        <svg
          className="mr-2"
          width="16"
          height="16"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 1.333l2.06 4.174 4.607.673-3.334 3.247.787 4.586L8 11.847l-4.12 2.166.787-4.586L1.333 6.18l4.607-.673L8 1.333z"
            fill="#FF6F41"
          />
        </svg>
        <p className="mr-1">{vote_average}</p>
        <p style={{ color: "rgba(255, 255, 255, 0.5)" }}>({popularity})</p>
      </div>
    </div>
  );
};

export default Card;