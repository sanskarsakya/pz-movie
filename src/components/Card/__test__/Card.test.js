import React from "react";
import ReactDOM from "react-dom";

import Card from "../Card";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);


describe("Card test suite", () => {

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<BrowserRouter><Card /></BrowserRouter>, div);
    });

    // it("renders card correctly", () => {
    //     const { getByTestId } = render(<Card name="terminator" />);
    //     expect(getByTestId("movie_name")).toHaveTextContent("terminator");
    // });

    // it("renders actor correctly", () => {
    //     const { getByTestId } = render(<Actor name="test name" />);
    //     expect(getByTestId("actor_name")).toHaveTextContent("test name");
    // });

    it("should match snapshot", () => {
        const tree = renderer.create(
        <BrowserRouter>
            <Card id="1" name="terminator" image="test image" vote_average="7.8" popularity="7" year="2020" />
        </BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
