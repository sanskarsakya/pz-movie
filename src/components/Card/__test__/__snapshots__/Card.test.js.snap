// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suite should match snapshot 1`] = `
<div
  className="w-1/5 p-3 transform hover:-translate-y-2 ease-in-out duration-200 transition-translate"
>
  <div
    className="overflow-hidden"
    style={
      Object {
        "borderRadius": 20,
        "height": 200,
      }
    }
  >
    <img
      alt=""
      className="object-cover"
      src="https://image.tmdb.org/t/p/w500test image"
    />
  </div>
  <a
    href="/movie/1"
    onClick={[Function]}
  >
    <p
      className="mt-5 text-xl font-semibold cursor-pointer"
      data-testid="movie_name"
    >
      terminator
    </p>
  </a>
  <div
    className="mt-1 text-sm tracking-wider flex items-center"
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.8)",
      }
    }
  >
    <span
      className="mr-6"
    >
      2020
    </span>
    <p>
      Adventure, Comedy
    </p>
  </div>
  <div
    className="mt-3 text-sm tracking-wider flex items-center"
  >
    <svg
      className="mr-2"
      fill="none"
      height="16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 1.333l2.06 4.174 4.607.673-3.334 3.247.787 4.586L8 11.847l-4.12 2.166.787-4.586L1.333 6.18l4.607-.673L8 1.333z"
        fill="#FF6F41"
      />
    </svg>
    <p
      className="mr-1"
    >
      7.8
    </p>
    <p
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.5)",
        }
      }
    >
      (
      7
      )
    </p>
  </div>
</div>
`;
